spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveViews.scala

```diff
@@ -0,0 +1,231 @@
+        .getOrElse(u)
+
+    case ShowCreateTable(ResolvedV2View(_, ident, view), _, _) =>
+      ShowCreateV2View(V2ViewDescription(ident.quoted, view))
+
+    case DescribeRelation(ResolvedV2View(_, ident, view), _, isExtended, _) =>
+      DescribeV2View(V2ViewDescription(ident.quoted, view), isExtended)
+
+    case ShowTableProperties(ResolvedV2View(_, ident, view), propertyKeys, _) =>
+      ShowV2ViewProperties(V2ViewDescription(ident.quoted, view), propertyKeys)
+
+    case SetViewProperties(ResolvedV2View(catalog, ident, _), props) =>
+      val changes = props.map {
+        case (property, value) => ViewChange.setProperty(property, value)
+      }.toSeq
+      AlterV2View(catalog, ident, changes)
+
+    case UnsetViewProperties(ResolvedV2View(catalog, ident, _), propertyKeys, ifExists) =>
+      if (!ifExists) {
+        val view = catalog.loadView(ident)
+        propertyKeys.filterNot(view.properties.containsKey).foreach { property =>
+          cannotUnsetNonExistentViewProperty(ident, property)
+        }
+      }
+      val changes = propertyKeys.map(ViewChange.removeProperty)
+      AlterV2View(catalog, ident, changes)
+
+    case RenameTable(ResolvedV2View(oldCatalog, oldIdent, _),
+    NonSessionCatalogAndIdentifier(newCatalog, newIdent), true) =>
+      if (oldCatalog.name != newCatalog.name) {
+        cannotMoveViewBetweenCatalogs(
+          oldCatalog.name, newCatalog.name)
+      }
+      RenameV2View(oldCatalog, oldIdent, newIdent)
+
+    case p@SubqueryAlias(_, view: IcebergView) =>
+      p.copy(child = resolveViews(view))
+
+  }
+
+  def cannotUnsetNonExistentViewProperty(ident: Identifier, property: String): Throwable =
+    throw new AnalysisException(
+      s"Attempted to unset non-existent property '$property' in view $ident")
+
+  def cannotMoveViewBetweenCatalogs(oldCatalog: String, newCatalog: String): Throwable =
+    throw new AnalysisException(
+      s"Cannot move view between catalogs: from=$oldCatalog and to=$newCatalog")
+
+  def lookupView(catalog: CatalogPlugin, ident: Identifier): Option[LogicalPlan] =
+    loadView(catalog, ident).map {
+      case view if CatalogV2Util.isSessionCatalog(catalog) =>
+        ResolvedPersistentView(catalog, ident, view.schema)
+      case view =>
+        ResolvedV2View(CatalogHelper(catalog).asViewCatalog, ident, view)
+    }
+
+  // The current catalog and namespace may be different from when the view was created, we must
+  // resolve the view logical plan here, with the catalog and namespace stored in view metadata.
+  // This is done by keeping the catalog and namespace in `AnalysisContext`, and analyzer will
+  // look at `AnalysisContext.catalogAndNamespace` when resolving relations with single-part name.
+  // If `AnalysisContext.catalogAndNamespace` is non-empty, analyzer will expand single-part names
+  // with it, instead of current catalog and namespace.
+  def resolveViews(plan: LogicalPlan): LogicalPlan = plan match {
```

Comment:
in https://github.com/apache/spark/pull/44197 resolving views happens in the `Analyzer`